test/functional/local_transactions_controller_test.rb

```diff
@@ -59,6 +59,30 @@ class LocalTransactionsControllerTest < ActionController::TestCase
       end
     end
 
+    context "loading the local transaction when posting an invalid postcode" do
+      setup do
+        mapit_does_not_have_a_bad_postcode("BLAH")
+
+        post :publication, slug: "send-a-bear-to-your-local-council", postcode: "BLAH"
+      end
+
+      should "log the invalid postcode error" do
+        assert_equal assigns(:postcode_error), "invalidPostcodeFormat"
+      end
+    end
+
+    context "loading the local transaction when posting a postcode with no matching areas" do
+      setup do
+        mapit_does_not_have_a_postcode("WC1E 9ZZ")
+
+        post :publication, slug: "send-a-bear-to-your-local-council", postcode: "WC1E 9ZZ"
+      end
+
+      should "log the invalid postcode error" do
+        assert_equal assigns(:postcode_error), "fullPostcodeNoMapitMatch"
+      end
+    end
```

Comment:
There's 3 possible postcode errors, but only 2 specs.  We might also want to explicitly test that we don't set postcode errors for non local transaction formats (this might be overkill, I'd be open to being talked down from that).